### ISSUE TITLE
URL Handling (For sharing)

### ISSUE THREAD

**Issue by pengyin-shan** [healthyregions/SDOHPlace#395]

**Comment by mradamcox** [comment]
Let's begin by listing everything we need... Here's `current name` / `potential new short name` for everything I can think of, organized by the actions needed. Keep in mind, now that a separate internal manager is used for state, the URL params are only used to set the initial state of the page, i.e. to show a user a shared version of the page. The URL params will be updated as the page state is changed, but they do not drive functionality after the initial page load.

### keep as is but maybe shorten name

- `query` / `q` : the string that appears in the search box and is the main query to solr
- `show` / `id` : id of the item to show for details
- `spatial_resolution` / `sr`
- `subject` / `su`

### add/change behavior of these params

- `bbox` / `bb` : bbox param used for "search this area". If present, that search filter will be enabled, and the map itself will be zoomed to this bounding box. If param is not present, the filter will not be enabled.
- `index_year` / `yr` : The behavior of this param should be changed so that it takes a range of years rather than a list.
  - For example `yr=2000-2005` would be translated within the function that generates the Solr query to a range query on the `index_year` field. For a single year, the same pattern, `yr=2000-2000` would be fine for now.
- `sortBy` & `sortOrder` > `so` : Combine these two parameters into a single `so` param.
  - `so=latest` : sorts records by index_year, most recent first (currently it looks like the "modified" field is used, but this is not a relevant field for data discovery)
  - `so=earliest` : sorts records by index_year, oldest first (currently it looks like the "modified" field is used, but this is not a relevant field for data discovery)
  - `so=??` : (we can introduce more keywords down the road for sort if necessary)
  - just like the current behavior, "relevance" is the default sort order if no param is present (or if `so` doesn't equal one of the preset 

### remove, no longer needed

- `sortBy` & `sortOrder` : now redundant, see above
- `filterPanel` : no need to be able to share this aspect of the page state
- `showInfoPanel` : no need to be able to share this aspect of the page state
- `showSharedLink` : no need to be able to share this aspect of the page state
- `bboxSearch` : now this will be implied by the presence or absence of the `bbox` param
- `layers` / `overlays` : I don't think the presence/absence of various map reference layers really needs to be shared or retained on initial page load. Perhaps, we can reintroduce this behavior later, only when `bbox`/`bb` is present

@pengyin-shan Are there parameters I'm missing here? This list is important to decide whether we hash the URL or not.

**Comment by pengyin-shan** [comment]
@mradamcox these should be current parameters. To confirm, by removing parameters such as filterPanel, showInfoPanel, and showSharedLink, users would see the default view (e.g., the filter panel is hidden, the detail panel is shown, etc.) when they copy and paste the current URL, rather than replicating the exact state of the page. Is this the intended behavior?

Alternatively, we could hash these attributes and create a parameter, such as st (for page state), to encapsulate them together.

**Comment by mradamcox** [comment]
Yes, that's correct, the params I listed as to be removed were concerned with aspects of the page state that I don't think need to be retained upon share (some never needed to be in the first place, like `showSharedLink`, for example).

**Comment by pengyin-shan** [comment]
We can do another round to shorten all of the parameters (like `query/q`), but I see that we have a significant reduction in parameter strings after the refactor. @mradamcox, what do you think about this? Do we still want to shorten all of them? It’s helpful to maintain the same url parameter names to match the state setup for easier code maintenance, but we can also change them all to abbreviations

**Comment by mradamcox** [comment]
I still kind of like the idea of shortening them, though I do see what you mean about matching the state setup. Let's just hang on to this idea for a little while longer, we'll make sure to resolve it one way or another by the end of January though.

**Comment by pengyin-shan** [comment]
> I still kind of like the idea of shortening them, though I do see what you mean about matching the state setup. Let's just hang on to this idea for a little while longer, we'll make sure to resolve it one way or another by the end of January though.

Sounds good. Let's do a final review after we finish adding all of the parameters. If we need to shorten the URL, we can do it all at once
